### ISSUE TITLE
Broken `config:dump-reference` command

### ISSUE THREAD

**Issue by gnutix** [Tony133/SymfonyFlexApiExample#1]
Hello,

I've noticed that the command `config:dump-reference` is broken because of the `yield` in https://github.com/Tony133/SymfonyFlexApiExample/blob/master/src/Kernel.php#L35, it throws an Exception with the message `Cannot traverse an already closed generator`.

The simple fix is to use a temporary array : 
```
        $bundles = [];
        $contents = require dirname(__DIR__).'/etc/bundles.php';
        foreach ($contents as $class => $envs) {
            if (isset($envs['all']) || isset($envs[$this->environment])) {
                $bundles[] = new $class();
            }
        }
        return $bundles;
```

But I would be curious if there's a "better way" ?
gnutix

**Comment by Tony133** [comment]
Hello,

I checked what he said but he does not launch any such exception. I tried to get the project up but I did not have any problem. I did not change the kernel.php file but left it by default.
Me ```config: dump-reference``` is not broken.
What kind of php version does it use?

I php 7.1.4

Try to do this:

Repo the repo, before making ```composer install``` make ```cp .env.dist .env``` later composer install then doing:

```
php bin/console config: dump-reference

```
or

```
./bin/console config: dump-reference

```

I attached the photos

<img width="903" alt="photo1" src="https://cloud.githubusercontent.com/assets/11542387/25772043/f3c866a6-3261-11e7-9e6a-5c599df11022.png">
<img width="943" alt="photo2" src="https://cloud.githubusercontent.com/assets/11542387/25772044/f6343aaa-3261-11e7-8517-b9a061bf294e.png">


Tony133
